crates/iceberg/src/spec/values.rs

```diff
@@ -893,9 +925,9 @@ impl Literal {
                 PrimitiveLiteral::Binary(_) => Type::Primitive(PrimitiveType::Binary),
                 PrimitiveLiteral::String(_) => Type::Primitive(PrimitiveType::String),
                 PrimitiveLiteral::UUID(_) => Type::Primitive(PrimitiveType::Uuid),
-                PrimitiveLiteral::Decimal(dec) => Type::Primitive(PrimitiveType::Decimal {
-                    precision: 38,
-                    scale: dec.scale(),
+                PrimitiveLiteral::Decimal(_) => Type::Primitive(PrimitiveType::Decimal {
+                    precision: MAX_DECIMAL_PRECISION,
+                    scale: 0,
```

Comment:
Are we ignoring the scale?

Reply:
How are making PrimitiveLiteral::Decimal as `(i128,i64) ` to store the value in the first one and the scale in the later one? 🤔

Reply:
When will this be used? I think inferring type from  literal is not feasible?

Reply:
> I think inferring type from literal is not feasible?
Sounds reasonable. In most of case we can get the type from the according schema. cc @JanKaul

Reply:
Initially I thought it would be useful to directly get the type from the value. But in case of the decimal you would need to store the Decimal with the scale like @ZENOTME suggested. It might make more sense to delete this method entirely and use the schema to get the types.